### PR TITLE
STITCH-2844 Swift SDK: Fix Update => Update, Update => Replace, Replace => Update Coalescence

### PR DESCRIPTION
**Pull request by jsflax**

### CODE REVIEW

File: Core/StitchCoreSDK/Tests/StitchCoreSDKTests/Auth/Internal/AccessTokenRefresherUnitTests.swift [hunk 1]
```diff
@@ -10,7 +10,9 @@ import SwiftJWT
 var ufreshJWT = JWT<ClaimsStandardJWT>.init(claims: ClaimsStandardJWT.init(exp: Date().addingTimeInterval(20 * 60)))
 let freshJWT = try! ufreshJWT.sign(using: JWTSigner.hs256(key: "secret".data(using: .utf8)!))
 
-var uexpiredJWT = JWT<ClaimsStandardJWT>.init(claims: ClaimsStandardJWT.init(exp: Date().addingTimeInterval(-(Date.init().timeIntervalSince1970 - 10000.0))))
+var uexpiredJWT = JWT<ClaimsStandardJWT>.init(
```
Comment (dkaminsky) [inline Core/StitchCoreSDK/Tests/StitchCoreSDKTests/Auth/Internal/AccessTokenRefresherUnitTests.swift]: formatting fix?

**Comment by jsflax** [comment]
Yep, quick drive by

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/CoreDocumentSynchronizationConfig.swift [hunk 2]
```diff
@@ -359,6 +359,44 @@ final class CoreDocumentSynchronization: Codable, Hashable {
...
+            default:
+                break
+            }
+        case .replace:
+            switch newestChangeEvent.operationType {
+            case .update:
+                return ChangeEvent(id: newestChangeEvent.id,
+                                   operationType: .replace,
+                                   fullDocument: newestChangeEvent.fullDocument,
+                                   ns: newestChangeEvent.ns,
```
Comment (tkaye407) [inline Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/CoreDocumentSynchronizationConfig.swift]: Is this only done on local changeEvents in which there is a fullDocument field?

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/UpdateDescription.swift [hunk 3]
```diff
@@ -44,7 +55,34 @@ public final class UpdateDescription: Codable {
...
+    }
+
+    /**
+     Unilaterally merge an update description into this update description.
+     - parameter otherDescription: the update description to merge into this
+     - returns: this merged update description
+     */
+    @discardableResult
+    public func merge(with otherDescription: UpdateDescription) -> UpdateDescription {
+        self.updatedFields = self.updatedFields.filter {
```
Comment (tkaye407) [inline Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/UpdateDescription.swift]: Is this necessary? Will line 75 not just overwrite it?

Comment (adamchel) [reply Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/UpdateDescription.swift]: this is different, this is to remove fields from updatedFields that were removed by the new update description. Line 75 adds the updates from the new update description while keeping the old ones that weren't part of the new removed fields

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/UpdateDescription.swift [hunk 4]
```diff
@@ -44,7 +55,34 @@ public final class UpdateDescription: Codable {
...
+        }
+        self._removedFields = NSOrderedSet(array: self.removedFields.filter {
+            return !otherDescription.updatedFields.keys.contains($0)
+        })
+
+        otherDescription.updatedFields.forEach {
+            self.updatedFields[$0.key] = $0.value
+        }
+
+        guard let mutableCopy = self._removedFields.mutableCopy() as? NSMutableOrderedSet else {
```
Comment (adamchel) [inline Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/UpdateDescription.swift]: [nit] can you add a comment explaining what's happening with this section of remaining code?

**Pull request closed without merging**